examples/infomax/infomax_route_example.cc

```diff
@@ -0,0 +1,150 @@
+
+using FloatType = float;
+using InfoMaxType = InfoMax<InSilicoRotater, FloatType>;
+
+void doTesting(const InfoMaxType &infomax, const std::vector<double> &allx,
+                 const std::vector<double> &ally, const std::vector<cv::Mat> &images)
+{
+    std::vector<double> x, y, u, v;
+    {
+        // Test network
+        std::cout << "Testing network..." << std::endl;
+        Timer<> t{ "Network testing took: " };
+        for (size_t i = 0; i < images.size(); i++) {
+            x.push_back(allx[i]);
+            y.push_back(ally[i]);
+
+            // Get heading and convert to vector
+            const radian_t heading = std::get<0>(infomax.getHeading(images[i]));
+            double sinx, cosx;
+            sincos(heading.value(), &sinx, &cosx);
+            u.push_back(cosx);
+            v.push_back(sinx);
+        }
+    }
+
+    // Output as quiver plot
+    std::map<std::string, std::string> kwargs;
+    kwargs["angles"] = "xy"; // This option is needed if you want matplotlib to do a sane plot :-/
+    plt::quiver(x, y, u, v, kwargs);
+    plt::xlabel("x (m)");
+    plt::ylabel("y (m)");
+    plt::plot({ x[0] }, { y[0] }, "b+");
+    plt::show();
+}
+
+int
+main(int argc, char **argv)
+{
+    if (argc < 2) {
+        std::cerr << "Error: must specify a route, e.g.:" << std::endl
+                  << "\t" << argv[0] << " ../../tools/ant_world_db_creator/ant1_route1 [num training repititions]" << std::endl;
+        return 1;
+    }
+
+    std::cout << "Eigen is using " << Eigen::nbThreads() << " threads." << std::endl
+              << std::endl;
+
+    // Where we are loading images from
+    const filesystem::path routePath(argv[1]);
+
+    // We resize images to this
+    const cv::Size imSize{ 90, 25 };
+
+    // Load images, resize and cache
+    std::vector<cv::Mat> images;
+    std::vector<double> x, y;
+    {
+        std::cout << "Loading images from " << routePath << "..." << std::endl;
+        Timer<> loadingTimer{ "Images loaded in: " };
+        const ImageDatabase routeImages(routePath);
+        assert(routeImages.size() > 0);
+
+        cv::Mat image;
+        for (size_t i = 1; i < routeImages.size(); i++) {
+            image = routeImages[i].loadGreyscale();
+            images.emplace_back(imSize, CV_8UC1);
+            cv::resize(image, images.back(), imSize);
+
+            // Save x and y in metres
+            x.push_back(routeImages[i].position[0].value() / 1000.0);
+            y.push_back(routeImages[i].position[1].value() / 1000.0);
+        }
+    }
+
+    // Number of times each image is presented to network
+    int numReps = (argc > 2) ? std::stoi(argv[2]) : 1;
+    const filesystem::path netPath = "trained_network_"s + routePath.filename()
+            + "_reps"s + std::to_string(numReps) + ".bin"s;
+
+    // If we already have a network for these params, load from disk
+    if (netPath.exists()) {
+        std::cout << "Loading weights from " << netPath << std::endl;
```

Comment:
I think this loading and saving logic belongs in the infomax class

Reply:
Ok good idea

Reply:
Although actually I am using this function to do things other than load weights. I use it in infomax_matlab_comparison to check the values of u and y etc.